apps/server/src/modules/files-storage/service/files-storage.service.ts

```diff
@@ -277,20 +277,29 @@ export class FilesStorageService {
 		targetParams: FileRecordParams
 	): Promise<CopyFileResponse[]> {
 		this.logger.debug({ action: 'copy', sourceFileRecords, targetParams });
-		const responseEntities: CopyFileResponse[] = [];
 
 		const promises = sourceFileRecords.map(async (sourceFile) => {
-			if (isStatusBlocked(sourceFile) || sourceFile.deletedSince) return;
+			if (isStatusBlocked(sourceFile)) return Promise.reject(new Error(ErrorType.FILE_IS_BLOCKED));
+			if (sourceFile.deletedSince) return Promise.reject(new Error(ErrorType.FILE_NOT_FOUND));
```

Comment:
@CeEv @SevenWaysDP Das deletedSince sollte aus meiner Sicht an dieser Stelle zumindest kein Error werfen bzw. sollte sich das copy dafür überhaupt interessieren? Wollen wir es an dieser Stelle ganz rausnehmen oder in ein Filter vor das map verlagern?

Reply:
Es soll tatsächlich kein Fehler werfen. Aber wir kopieren nicht die Dateien die als Blocked oder zum Löschen markiert sind. So wie es davor war schon korrekt.

Reply:
Um einen consistent return zu haben, können wir hier nicht undefined returnen. Deswegen geht es so wie vorher nicht mehr. Wenn deletedSince hier berücksichtigt werden soll, dann wäre ein filter vor dem map meine vorgeschlagene Lösung. Auch wenn wir dann eine Schleife mehr haben.

Reply:
Ah und Motivation für den Umbau ist etwas zu returnen und nicht mehr in das Array zu pushen.